client/finality-grandpa/src/finality_proof.rs

```diff
@@ -609,18 +611,22 @@ pub(crate) mod tests {
 			&self,
 			hash: H256,
 			header: Header,
-			proof: StorageProof
+			proof: StorageProof,
 		) -> ClientResult<AuthorityList> {
 			self.0(hash, header, proof)
 		}
 	}
 
 	#[derive(Debug, PartialEq, Encode, Decode)]
-	pub struct TestJustification(pub bool, pub Vec<u8>);
+	pub struct TestJustification(pub bool, pub Option<(u64, Vec<(AuthorityId, u64)>)>, pub Vec<u8>);
```

Comment:
Is this just a `Some` because you don't want to go through all the tests and give a value? Any reason not to make it required?

Reply:
Yes - that's the reason.

Reply:
I feel like there's not so many uses that it would be difficult to add the authority set and that's a good additional check to have in place anyway. And you could drop the first `bool` arg since `false` is the same as providing an incorrect set `id`.